modules/product/src/Plugin/Field/FieldFormatter/ProductAttributesOverview.php

```diff
@@ -0,0 +1,248 @@
+    }
+
+    $form['attributes'] = [
+      '#type' => 'checkboxes',
+      '#title' => $this->t('Render and display the following attributes.'),
+      '#default_value' => $this->getSetting('attributes'),
+      '#options' => $used_attributes,
+    ];
+
+    $attribute_value_view_modes = $this->entityDisplayRepository->getViewModes('commerce_product_attribute_value');
+
+    $form['view_mode'] = [
+      '#type' => 'select',
+      '#title' => $this->t('Display mode to render the attributes with'),
+      '#default_value' => $this->getSetting('view_mode'),
+      '#options' => [
+        'default' => $this->t('Default'),
+      ],
+      '#disabled' => empty($attribute_value_view_modes),
+    ];
+
+    foreach ($attribute_value_view_modes as $key => $value) {
+      $form['view_mode']['#options'][$key] = $value['label'];
+    }
+
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function settingsSummary() {
+    $product_attribute_storage = $this->entityTypeManager->getStorage('commerce_product_attribute');
+    $attributes = [];
+    if (empty($this->getSetting('attributes'))) {
+      $attributes[] = $this->t('None');
+    }
+    else {
+      foreach (array_filter($this->getSetting('attributes')) as $item) {
+        $attribute = $product_attribute_storage->load($item);
+        $attributes[] = $attribute->label();
+      }
+    }
+    $summary[] = $this->t('Displaying the following attributes: @attributes', [
+      '@attributes' => implode(', ', $attributes),
+    ]);
+    $summary[] = $this->t('Attribute value display mode: @mode', [
+      '@mode' => $this->getSetting('view_mode'),
+    ]);
+
+    return $summary;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function viewElements(FieldItemListInterface $items, $langcode) {
+    $product_attribute_storage = $this->entityTypeManager->getStorage('commerce_product_attribute');
+    $elements = [];
+    foreach (array_filter($this->getSetting('attributes')) as $item) {
+      $attribute = $product_attribute_storage->load($item);
+      $elements[] = $this->getAttributeItemList($items, $attribute);
+    }
+    return $elements;
+  }
+
+  /**
+   * Gets the renderable item list of attributes.
+   *
+   * @param \Drupal\Core\Field\FieldItemListInterface $variations
+   *   The item list of variation entities.
+   * @param \Drupal\commerce_product\Entity\ProductAttributeInterface $attribute
+   *   The product attribute.
+   *
+   * @return array
+   *   The render array.
+   */
+  protected function getAttributeItemList(FieldItemListInterface $variations, ProductAttributeInterface $attribute) {
+    $build = [
+      '#theme' => 'item_list',
+      '#title' => $attribute->label(),
+      '#items' => [],
+    ];
+
+    $view_builder = $this->entityTypeManager->getViewBuilder('commerce_product_attribute_value');
+
+    /** @var \Drupal\Core\Field\Plugin\Field\FieldType\EntityReferenceItem $variation */
+    foreach ($variations as $variation) {
```

Comment:
These are not really $variations if they're EntityReferenceItem instances. Let's do $variation_items